app/helpers/cms/url_helper.rb

```diff
@@ -21,12 +21,8 @@ def cms_published_url(page)
     }.freeze
 
     def cms_uri(page)
-      uri = URI.parse(access_code_url)
-      uri.host = page.provider.external_domain
-      uri.query = { return_to: page.path ? page.path : HARD_WIRED_PATHS[page.system_name],
-                    access_code: current_account.site_access_code,
-                    cms_token: page.provider.settings.cms_token! }.to_query
-      uri.port = request.port if Rails.env.development?
+      uri = URI.parse(provider_admin_cms_visit_portal_path)
+      uri.query = { return_to: page.path ? page.path : HARD_WIRED_PATHS[page.system_name]}.to_query
```

Comment:
I can't test this now. But when you remove also the `access_code`, wont that break the visit without edit use case?

Reply:
No, because here the URL points to `VisitPortalController` which receives other parameters. Then is the visit portal controller who redirects to `AccessCodesController` and sends the same parameters that were previously here, like the access code.

In the end, The idea is to prevent the UI elements like links or buttons from pointing to a URL which includes a signature that expires in a minute, that's why I placed the visit portal controller in the middle.